pennylane_qiskit/runtime.py

```diff
@@ -0,0 +1,257 @@
+
+        if self.kwargs.get("init_qubits"):
+            program_inputs["init_qubits"] = self.kwargs.get("init_qubits")
+
+        if self.kwargs.get("rep_delay"):
+            program_inputs["rep_delay"] = self.kwargs.get("rep_delay")
+
+        if self.kwargs.get("transpiler_options"):
+            program_inputs["transpiler_options"] = self.kwargs.get("transpiler_options")
+
+        program_inputs["measurement_error_mitigation"] = self.kwargs.get(
+            "measurement_error_mitigation", False
+        )
+
+        # Specify the backend.
+        options = {"backend_name": self.backend.name()}
+
+        # Send circuits to the cloud for execution by the circuit-runner program.
+        job = self.provider.runtime.run(
+            program_id="circuit-runner", options=options, inputs=program_inputs
+        )
+        self._current_job = job.result(decoder=RunnerResult)
+
+        results = []
+
+        for index, circuit in enumerate(circuits):
+            self._samples = self.generate_samples(index)
+            res = self.statistics(circuit.observables)
+            results.append(res)
+
+        if self.tracker.active:
+            job_time = {
+                "total_time": self._current_job._metadata.get("time_taken"),
+            }
+            self.tracker.update(batches=1, batch_len=len(circuits), job_time=job_time)
+            self.tracker.record()
+
+        return results
+
+    def generate_samples(self, circuit=None):
+        r"""Returns the computational basis samples generated for all wires.
+
+        Note that PennyLane uses the convention :math:`|q_0,q_1,\dots,q_{N-1}\rangle` where
+        :math:`q_0` is the most significant bit.
+
+        Args:
+            circuit (int): position of the circuit in the batch.
+
+        Returns:
+             array[complex]: array of samples in the shape ``(dev.shots, dev.num_wires)``
+        """
+        counts = self._current_job.get_counts()
+
+        # Batch of circuits
+        if not isinstance(counts, dict):
+            counts = self._current_job.get_counts()[circuit]
+
+        samples = []
+        for key, value in counts.items():
+            for _ in range(0, value):
+                samples.append(key)
+        return np.vstack([np.array([int(i) for i in s[::-1]]) for s in samples])
+
+
+class IBMQSamplerDevice(IBMQDevice):
+    r"""Class for a Qiskit runtime sampler program device in PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``).
+        provider (Provider): The Qiskit simulation provider
+        backend (str): the desired backend
+        shots (int or None): number of circuit evaluations/random samples used
+            to estimate expectation values and variances of observables. For statevector backends,
```

Comment:
From what I am seeing, the `IBMQSampler` does not make use of the number of shots (you can check it by setting `shots = 1`). Therefore I would remove it from the arguments.

Reply:
Thank you, good catch! Shots attribute is passed through the `run_config` kwarg.